### PR TITLE
Reorganize docs before update from work on Notion.

### PR DESCRIPTION
**Pull request by fmassot**
Don't look at the context, this PR targets first a reorganization of the docs layout.

### CODE REVIEW

File: docs/design/architecture.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+---
+title: Architecture
+sidebar_position: 1
+---
+
+Quickwit is built on three pillars:
+- the index: a set of data structures tailored to be queried, even on high latency storage;
```
Comment (PSeitz) [inline docs/design/architecture.md:7]:
```suggestion
- the index: a set of data structures tailored to be queried on high latency storage;
```

File: docs/design/architecture.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
+---
+title: Architecture
+sidebar_position: 1
+---
+
+Quickwit is built on three pillars:
```
Comment (PSeitz) [inline docs/design/architecture.md:6]:
```suggestion
Search in Quickwit  is built on three pillars:
```
without the search part this list feels a little incoherent

File: docs/design/architecture.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+![Quickwit Architecture](../assets/images/quickwit-architecture.svg)
+
+</div>
+
+## The index
+
+A quickwit index stores documents and makes it possible to query them efficiently.
+The index organizes documents into a collection of smaller independent indexes called splits.  Behind the scenes, a split is just a [tantivy index](https://github.com/tantivy-search/tantivy/blob/main/ARCHITECTURE.md#index-and-segments) with a custom index format.
+
+You have total control over how document fields are stored and indexed, thanks to the `Index config`.
```
Comment (PSeitz) [inline docs/design/architecture.md:25]: add link to index config

File: docs/design/architecture.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+### Index config
+
+A document is a collection of fields and associated values. Fields can be stored in different data structures:
+- an inverted index, which enables fast full-text search;
+- a column-oriented storage called `fast field`, which is the equivalent of `DocValues` in Lucene. Fast fields are required for computing aggregates over documents matching a query. They also allow some advanced types of filtering;
+- a row-store, called the `doc store` making it possible to retrieve the content of matching documents.
```
Comment (PSeitz) [inline docs/design/architecture.md:32]:
```suggestion
- The `doc store`, a compressed a row-store, making it possible to retrieve the content of matching documents.
```

File: docs/design/architecture.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+The index config controls how to map JSON objects to Quickwit documents and, for each field, defines whether it should be stored, indexed, or encoded as a fast field.
```
Comment (PSeitz) [inline docs/design/architecture.md:34]:
```suggestion
The index config controls how to map JSON objects to Quickwit documents and defines for each field whether it should be stored, indexed, or encoded as a fast field.
```

File: docs/design/architecture.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+You can define one with a `JSON file`, [see docs](../reference/index-config.md).
```
Comment (PSeitz) [inline docs/design/architecture.md:35]: yaml is preferred now?

File: docs/design/architecture.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+### Splits
+
+A split is a small piece of an index identified by a UUID. For each split, Quickwit creates a `hotcache` file along with the index files. This hotcache enables the search nodes to open a split in less than 60ms, even on high latency storage.
```
Comment (PSeitz) [inline docs/design/architecture.md:39]: Maybe mention that a split is stored in

indexes/{index-id}/{split-id}.split

File: docs/design/architecture.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+Index metadata needs to be accessible by every instance of the cluster. This is made possible thanks to the `metastore`.
+
+
+## The metastore
+
+Quickwit gathers index metadata into a metastore to make them available across the cluster.
+
+For a given query on a given index, a search node will ask the metastore for the index metadata and then use it to do the query planning and finally execute the plan.
+
+Currently, the only implementation of the `metastore` is a JSON file based store: it writes a `quickwit.json` on disk or in an Amazon S3 bucket. We plan on supporting other backends such as Postgresql and other popular databases soon.
```
Comment (PSeitz) [inline docs/design/architecture.md:56]: we support already postgresql?

File: docs/design/architecture.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+- Returns aggregated results.
+
+### Cluster discovery
+
+Quickwit uses a gossip protocol to manage membership and broadcast messages to the cluster provided by [artillery project](https://github.com/bastion-rs/artillery/). The gossip protocol is based on "SWIM: Scalable Weakly-consistent Infection-style Process Group Membership Protocol" with a few minor adaptations.
+
+
+### Rendezvous hashing
+
+The root node uses [Rendezvous hashing](https://en.wikipedia.org/wiki/Rendezvous_hashing) to distribute the workload among leaf nodes. Rendez-vous hashing makes it possible to define a node/split affinity function with excellent stability properties when a node joins or leaves the cluster. This trick unlocks efficient caching. In Quickwit v0.1, however, caching is limited to the hotcache files.
```
Comment (PSeitz) [inline docs/design/architecture.md:89]: mentioning quickwit v0.1

File: docs/design/querying.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
+---
+title: Querying
+sidebar_position: 3
+---
+
+Quickwit is built on three pillars:
```
Comment (PSeitz) [inline docs/design/querying.md:6]: same as architecture.md?

File: docs/design/architecture.md (hunk 8 above)
Comment (PSeitz) [reply docs/design/architecture.md:56]: metastore.json not quickwit.json

File: docs/reference/index-config.md [hunk 11]
```diff
@@ -1,53 +1,73 @@
...
 
-This config can be expressed as a json file given to the `new` Quickwit command. Here is a example of a json config for a logging dataset:
+The index config lets you define four things:
+- The document model version, only version `0` is available.
+- The doc mapping, that is how a document and the fields it contains are stored and indexed for a given index.
+- The indexing settings that defines the merge policy, timestamp field used for sharding, and more.
+- The search settings that defines the default search fields `default_search_fields`, a list of field that will be used by default for search.
+- The (data) sources that defines a list of source of types among Kafka or simple files.  
+
+Configuration is set at index creation or can be modified later by using CLI commands or by directly changing settings in the metastore except for the doc mapping (or schema), which is today immutable.
```
Comment (PSeitz) [inline docs/reference/index-config.md:15]: I think only sources can be changed later via CLI

File: docs/reference/index-config.md [hunk 12]
```diff
@@ -181,15 +199,44 @@ Quickwit supports nested object as long as it does not contains arrays of object
...
 
 
-## Behaviour with fields are not defined in the config
+### Behaviour with fields are not defined in the config
 Fields in your json document that are not defined in the `index config` will be ignored.
 
 
-## Behaviour with null values or missing fields
+### Behaviour with null values or missing fields
 Fields with `Null` or missing field in your json document will be silently ignored when indexing.
```
Comment (PSeitz) [inline docs/reference/index-config.md:213]:
```suggestion
Fields with `null` or missing field in your json document will be silently ignored when indexing.
```

File: docs/reference/index-config.md [hunk 13]
```diff
@@ -181,15 +199,44 @@ Quickwit supports nested object as long as it does not contains arrays of object
...
+
+
+## Indexing settings
+
+This section describes indexing settings.
+
+| Variable      | Description   | Default value |
+| ------------- | ------------- | ------------- |
+| timestamp_field    | timestamp field `timestamp_field` used for [sharding documents in splits](../overview/architecture.md#splits). | None |
+| commit_timeout_secs | Maximum number of seconds before committing a split since its creation. | 60 |
```
Comment (PSeitz) [inline docs/reference/index-config.md:224]: I think potential costs should be mentioned here (S3 put requests)

File: docs/reference/index-config.md [hunk 14]
```diff
@@ -181,15 +199,44 @@ Quickwit supports nested object as long as it does not contains arrays of object
...
+## Search settings
+
+| Variable      | Description   | Default value |
+| ------------- | ------------- | ------------- |
+| search_default_fields | Default list of fields that will be used for search. | None |
+
+
+## Sources
+
+One index can define a list of sources, go to [sources] page to learn how to configure them.
```
Comment (PSeitz) [inline docs/reference/index-config.md:242]: add link to sources

File: docs/design/architecture.md (hunk 2 above)
Comment (fmassot) [reply docs/design/architecture.md:6]: yes this needs a complete rewrite. I just moved the files for now and everything else is on notion

File: docs/design/querying.md (hunk 10 above)
Comment (fmassot) [reply docs/design/querying.md:6]: sorry, I made some git mv and copy this one... I'm cleaning this now.

File: docs/reference/index-config.md (hunk 13 above)
Comment (fmassot) [reply docs/reference/index-config.md:224]: Will report that on notion as this part is obsolete now.

**Comment by fmassot** [comment]
thanks @PSeitz for your comments. I will report them when possible but I forgot to push all my commits I had locally :/

File: docs/reference/index-config.md (hunk 11 above)
Comment (fmassot) [reply docs/reference/index-config.md:15]: Yes indeed, I will update that on notion.

File: docs/reference/index-config.md (hunk 12 above)
Comment (fmassot) [reply docs/reference/index-config.md:213]: updated on Notion.

**Pull request merged**